### DISCUSSION TITLE
RABIES failed: Workflow did not execute cleanly

### DISCUSSION THREAD

**Discussion by MrImaging** [CoBrALab/RABIES#371] [Q&A]
Dear RABIES experts and colleagues,

First and foremost thank you for developing this great tool.

While I can sucessfully run the preprocessing in Docker on the test dataset (see code below) I am experiencing issues when trying to do so with my own data (it seems like the pipeline always crashes during antsRegistration_affine_SyN.sh):

_RABIES failed: Workflow did not execute cleanly. Check log for details_

**Proprocess test data** 
`docker run -it --rm -v $PWD/test_dataset:/input_BIDS:ro -v $PWD/test_dataset_preprocess_outputs:/preprocess_outputs/ ghcr.io/cobralab/rabies:latest -p MultiProc preprocess /input_BIDS/ /preprocess_outputs/ --commonspace_reg masking=true,brain_extraction=false,template_registration=SyN,fast_commonspace=false`

**Preprocess my data** 
`docker run -it --rm -v $PWD/MyData-test_Rabies:/input_BIDS:ro -v $PWD/MyData-test_Rabies_preprocess_outputs:/preprocess_outputs/ ghcr.io/cobralab/rabies:latest -p MultiProc preprocess /input_BIDS/ /preprocess_outputs/ --commonspace_reg masking=true,brain_extraction=false,template_registration=SyN,fast_commonspace=false`

My data was recorded on a Bruker scanner, using ParaVision V3.3, exportet as Dicom and converted to NIfTI using Chris Rorden's dcm2niix. The problem occures independently of the images' orientation with respect to the template brain. 

Any help or suggestions are highly appreciated.
Thanks!

BR
Peter 

Attachments
[rabies_preprocess.log](https://github.com/CoBrALab/RABIES/files/15306939/rabies_preprocess.log)

THE QC Report folder only contains template_files.png

**Comment by gdevenyi** [comment]
> My data was recorded on a Bruker scanner, using ParaVision V3.3, exportet as Dicom and converted to NIfTI using Chris Rorden's dcm2niix. The problem occures independently of the images' orientation with respect to the template brain.

We strongly recommend brkraw as the converter as it properly maintains coordinate information, while bruker's DICOM export does not.

Can you please provide some more details about the kinds of scans you are attempting to use.

**Reply by MrImaging** [reply]
Thank you for the quick reply. 
My subset for testing includes two subjects, with one session each, containing one coronal T2-weighted structural image (256x256x35) and one BOLD fMRI scan (96x96x35x480, TR = 1.5 s). If I can provide you with any specific sequence parameters, please let me know. 

I am trying to get brkraw to work to see whether this issue is related to some conversion problem, but unfortunately I am currently experiencing issues with it as well:

```
import sys
import numpy as np
np.set_printoptions(precision=3, suppress=True)
import brkraw
import pprint as pp

path = 'brukerFile.zip'
pvdset = brkraw.load(path)
pvdset.info()
```
```
pvdset.info()
    ^^^^^^^^^^^
AttributeError: 'BrukerLoader' object has no attribute 'info'
```

**Reply by gdevenyi** [reply]
I have only used brkraw in the terminal, if you require support I suggest using the repository issues/discussions for them.

**Comment by gdevenyi** [comment]
Examining your log files, other things of note:
```
/opt/conda/bin/antsRegistration_affine_SyN.sh: line 703: 18381 Killed                  ${run_command}
2024-05-14 09:41:42 UTC [    alert] Failed at 702: ${run_command}
```
This is the error that stopped the pipeline. You ran out of either memory or time and your job management system killed the process, this is not a RABIES issue.

Otherwise, I also see you only provided a single subject for the process. RABIES is intended to be used for cohort processing, you should provide at least 2 subjects.

**Reply by MrImaging** [reply]
Thank you for looking into this. 
I have tried cropping the time series to only 15 frames and monitored the available resources during execution. As this led to the same result, I'm inclined to think this is not caused by a memory issue. Nevertheless, I will check on a more powerful setup (and I am also still planning to try brkraw). 

I am almost positive to have provided structural and functional data of one session for two subjects (just like in the test data set) which I also see listed in the beginning of the log:

input_BIDS/
sub-F001/ 
ses-m002/ 
anat/ 
sub-F001_ses-m002_acq-cor_T2w.nii 
func/ 
sub-F001_ses-m002_task-rest_acq-EPI_bold.nii 
    
sub-F002/ 
ses-m002/ 
anat/ 
sub-F002_ses-m002_acq-cor_T2w.nii 
func/ 
sub-F002_ses-m002_task-rest_acq-EPI_bold.nii 

May I ask where in the the log the suggested 'loss' of a data-set could be seen?

**Reply by gdevenyi** [reply]
> , I'm inclined to think this is not caused by a memory issue.

I don't know the system you're running on, but the "KILLED" note is an indication that an external signal stopped the process. This could be out-of-memory, or it could be a maximum CPU time limit. Contact your administrator for support if this system is not yours.

> I am almost positive to have provided structural and functional data

My bad I misread the log.

**Reply by MrImaging** [reply]
Thank you for your insights. I have not yet been able to pinpoint the issue, but managed to run rabies successfully on another (much more powerful) system with data converted using brkraw.
